dataset/src/main/kotlin/com/cosmotech/dataset/service/DatasetServiceImpl.kt

```diff
@@ -863,9 +853,15 @@ class DatasetServiceImpl(
       workspaceId: String
   ): Dataset {
     val dataset = findDatasetById(organizationId, datasetId)
-    dataset.linkedWorkspaceIdList?.remove(workspaceId)
```

Comment:
doesn't need to check if object exist before remove

or keep it if it's to save a 'save'. but i don't think that's necessary

Reply:
Yup, the `contains` call is here just to not save dataset if it's not necessary.